exchanges/bybit/bybit_wrapper.go

```diff
@@ -2092,66 +1400,89 @@ func (by *Bybit) GetHistoricCandlesExtended(ctx context.Context, pair currency.P
+	return info.TimeNano.Time(), err
 }
 
 // UpdateOrderExecutionLimits sets exchange executions for a required asset type
 func (by *Bybit) UpdateOrderExecutionLimits(ctx context.Context, a asset.Item) error {
-	var limits []order.MinMaxLevel
+	avail, err := by.GetAvailablePairs(a)
+	if err != nil {
+		return err
+	}
+	var instrumentsInfo *InstrumentsInfo
 	switch a {
-	case asset.Spot:
-		pairsData, err := by.GetAllSpotPairs(ctx)
+	case asset.Spot, asset.USDTMarginedFutures, asset.USDCMarginedFutures, asset.CoinMarginedFutures:
+		instrumentsInfo, err = by.GetInstruments(ctx, getCategoryName(a), "", "", "", "", 400)
 		if err != nil {
 			return err
 		}
-
-		limits = make([]order.MinMaxLevel, 0, len(pairsData))
-		for x := range pairsData {
-			var pair currency.Pair
-			var enabled bool
-			pair, enabled, err = by.MatchSymbolCheckEnabled(pairsData[x].Name, a, false)
-			if err != nil {
-				log.Warnf(log.ExchangeSys, "%s unable to load limits for %v, pair data missing", by.Name, pairsData[x].Name)
-				continue
-			}
-			if !enabled {
-				continue
-			}
-
-			limits = append(limits, order.MinMaxLevel{
-				Asset:                   a,
-				Pair:                    pair,
-				AmountStepIncrementSize: pairsData[x].BasePrecision.Float64(),
-				QuoteStepIncrementSize:  pairsData[x].QuotePrecision.Float64(),
-				MinimumBaseAmount:       pairsData[x].MinTradeQuantity.Float64(),
-				MaximumBaseAmount:       pairsData[x].MaxTradeQuantity.Float64(),
-				MinimumQuoteAmount:      pairsData[x].MinTradeAmount.Float64(),
-				MaximumQuoteAmount:      pairsData[x].MaxTradeAmount.Float64(),
-				PriceStepIncrementSize:  pairsData[x].MinPricePrecision.Float64(),
-			})
+	case asset.Options:
+		instrumentsInfo, err = by.GetInstruments(ctx, getCategoryName(a), "", "", "BTC", "", 400)
+		if err != nil {
+			return err
 		}
 	default:
-		// TODO: Add in other assets
 		return fmt.Errorf("%s %w", a, asset.ErrNotSupported)
 	}
+	limits := make([]order.MinMaxLevel, 0, len(instrumentsInfo.List))
+	for x := range instrumentsInfo.List {
+		var pair currency.Pair
+		pair, err = avail.DeriveFrom(instrumentsInfo.List[x].Symbol)
+		if err != nil {
+			log.Warnf(log.ExchangeSys, "%s unable to load limits for %v, pair data missing", by.Name, instrumentsInfo.List[x].Symbol)
+			continue
+		}
+
+		limits = append(limits, order.MinMaxLevel{
+			Asset:                   a,
+			Pair:                    pair,
+			MinimumBaseAmount:       instrumentsInfo.List[x].LotSizeFilter.MinOrderQty.Float64(),
+			MaximumBaseAmount:       instrumentsInfo.List[x].LotSizeFilter.MaxOrderQty.Float64(),
+			MinPrice:                instrumentsInfo.List[x].PriceFilter.MinPrice.Float64(),
+			MaxPrice:                instrumentsInfo.List[x].PriceFilter.MaxPrice.Float64(),
+			PriceStepIncrementSize:  instrumentsInfo.List[x].PriceFilter.TickSize.Float64(),
+			AmountStepIncrementSize: instrumentsInfo.List[x].LotSizeFilter.QtyStep.Float64(),
```

Comment:
These are purging a lot of values. 
```
{"symbol":"BTCUSDT","baseCoin":"BTC","quoteCoin":"USDT","innovation":"0","status":"Trading","marginTrading":"both","lotSizeFilter":{"basePrecision":"0.000001","quotePrecision":"0.00000001","minOrderQty":"0.000048","maxOrderQty":"71.73956243","minOrderAmt":"1","maxOrderAmt":"2000000"},"priceFilter":{"tickSize":"0.01"}}
```

```
{Pair:BTC:USDT Asset:spot MinPrice:0 MaxPrice:0 PriceStepIncrementSize:0.01 MultiplierUp:0 MultiplierDown:0 MultiplierDecimal:0 AveragePriceMinutes:0 MinimumBaseAmount:4.8e-05 MaximumBaseAmount:71.73956243 MinimumQuoteAmount:0 MaximumQuoteAmount:0 AmountStepIncrementSize:0 QuoteStepIncrementSize:0.01 MinNotional:0 MaxIcebergParts:0 MarketMinQty:0 MarketMaxQty:0 MarketStepIncrementSize:0 MaxTotalOrders:0 MaxAlgoOrders:0}
```

AmountStepIncrementSize should be: instrumentsInfo.List[x].LotSizeFilter.BasePrecision.Float64()


For min and max quote amountsit seems we don't need to multiply by price?

Also `QuoteStepIncrementSize:  instrumentsInfo.List[x].LotSizeFilter.QuotePrecision.Float64(),` seems to be more correct could you please recheck?

Reply:
I hope I have resolved this part.